services/contract_call_local.proto

```diff
@@ -162,6 +162,12 @@ message ContractFunctionResult {
      * ContractCreateTransactionBody or a ContractCallTransactionBody.     
      */
     AccountID sender_id = 13;
+
+    /**
+    * Map used to track contract nonces for all contracts that have participated in the transaction.
+    * The keys would have the different contract IDs as strings in "shard.realm.num" format and the values are the new nonce values.
+    */
+    map<string, int64> contract_nonces = 14;
```

Comment:
Instead of providing the contract ID as a string and having to parse it by each client, an alternative solution will be to create a new type:
```
message ContractNonceInfo {
  ContractID contract_id = 1;
  int64 nonce = 2;
}
```

and then add `repeated ContractNonceInfo = 14` to the ContractFunctionResult type.

Reply:
I think Stoyan's idea is the better approach here.

Reply:
Updated.